src/components/SelectSearch.tsx

```diff
@@ -105,35 +106,53 @@ export default class SelectSearch extends React.PureComponent<
 
     private loadOptions(input: string, callback: Function): void {
         if (!this.props.url) {
-            callback(this.search(input));
+            const options: SelectSearchResult = this.search(input);
+            callback(options);
         } else {
             axios.get(this.props.url).then((response: AxiosResponse) => {
-                const results: SearchResult[] = [];
-                response.data.results.forEach((result: any) =>
-                    results.push({
-                        name: result.name,
-                        id: result.uuid,
-                        type: this.props.resultType
+                const results: SearchResult[] = response.data.results.map(
+                    ({ name, uuid, type }: any) => ({
+                        name,
+                        id: uuid,
+                        type
                     })
                 );
-                callback(null, this.search(input, results));
+                const options: SelectSearchResult = this.search(input, results);
+                callback(null, options);
             });
         }
     }
 
-    private onChange(selection: any): void {
-        if (!this.props.multi) {
-            selection = [selection];
+    // If 'multi' prop is truthy, we get an array. If not, we get a single object.
+    private onChange(selections: SearchResult | SearchResult[]): void {
```

Comment:
wanted to be explicit in accounting for the types of values this method could receive